modules/hcp-ibm-power-create-heterogeneous-nodepools-agent-hc.adoc

```diff
@@ -0,0 +1,201 @@
++
+[source,terminal]
+----
+$ export CLUSTERS_NAMESPACE=<hosted_cluster_namespace>
+$ export HOSTED_CLUSTER_NAME="<hosted_cluster_name>”
+$ export HOSTED_CONTROL_PLANE_NAMESPACE="${CLUSTERS_NAMESPACE}-${HOSTED_CLUSTER_NAME}"
+$ export BASEDOMAIN=<basedomain>
+$ export PULL_SECRET_FILE=$PWD/pull-secret
+$ export OCP_RELEASE=<release_image_version>
+$ export MACHINE_CIDR=192.168.122.0/24
+# Typically the namespace is created by the hypershift-operator 
+# but agent cluster creation generates a capi-provider role that
+# needs the namespace to already exist
+$ oc create ns ${HOSTED_CONTROL_PLANE_NAMESPACE}
+
+$ hypershift create cluster agent \
+    --name=${HOSTED_CLUSTER_NAME} \
+    --pull-secret=${PULL_SECRET_FILE} \
+    --agent-namespace=${HOSTED_CONTROL_PLANE_NAMESPACE} \
+    --base-domain=${BASEDOMAIN} \
+    --api-server-address=api.${HOSTED_CLUSTER_NAME}.${BASEDOMAIN} \
+    --release-image=quay.io/openshift-release-dev/ocp-release:${OCP_RELEASE}
+----
+
+
+[id="hcp-create-heterogeneous-nodepools_{context}"]
+== Create Heterogeneous NodePools
+
+By default, the previous section creates a nodepool on x86_64 architecture, to create heterogeneous nodepools, you can apply the below manifest with the architecture you want.
+
+.Procedure
+
+. to create heterogeneous nodepools
++
+[source,terminal]
+----
+apiVersion: hypershift.openshift.io/v1beta1
+kind: NodePool
+metadata:
+  name: ${HOSTED_CLUSTER_NAME}
+  namespace: ${CLUSTERS_NAMESPACE}
+spec:
+  arch: ${ARCH_PPC64LE}
+  clusterName: ${HOSTED_CLUSTER_NAME}
+  management:
+    autoRepair: false
+    upgradeType: InPlace
+  nodeDrainTimeout: 0s
+  nodeVolumeDetachTimeout: 0s
+  platform:
+    agent:
+      agentLabelSelector:
+        matchLabels:
+          inventory.agent-install.openshift.io/cpu-architecture: ${ARCH_PPC64LE}
+    type: Agent
+  release:
+    image: quay.io/openshift-release-dev/ocp-release:${OCP_RELEASE}
+  replicas: 0
+----
++
+This will create a nodepool of architecture ppc64le with zero replicas.
++
+[source,terminal]
+----
+agentLabelSelector:
+    matchLabels:
+        inventory.agent-install.openshift.io/cpu-architecture: ppc64le
+----
++
+This selector block is used to select the agents which are matching the given label. Which will ensure it will select only the agents from ppc64le architecture when it is scaled.
+
+
+[id="hcp-create-infraenv_{context}"]
+== Create Infraenv
+
+For heterogeneous nodepools, you need to create infraenv for each architecture you are going to have in your HCP. i.e. for heterogeneous nodepools with x86_64 and ppc64le architecture, you will need to create two infraenvs with both architectures.
+
+.Procedure
+
+. Create Infraenv
++
+[source,terminal]
+----
+export SSH_PUB_KEY=$(cat $HOME/.ssh/id_rsa.pub)
+----
++
+[source,terminal]
+----
+envsubst <<"EOF" | oc apply -f -
+apiVersion: agent-install.openshift.io/v1beta1
+kind: InfraEnv
+metadata:
+  name: ${HOSTED_CLUSTER_NAME}-${ARCH_X86}
+  namespace: ${HOSTED_CONTROL_PLANE_NAMESPACE}
```

Comment:
```suggestion
  namespace: <hosted_cluster_name>-<arch_x86>
```
You can convert all of the environment variables into user-replaced values (provided there is no mention of `export` commands for env vars). i.e user will replace variables with their custom values. See this guideline to know how to use/write them: https://redhat-documentation.github.io/supplementary-style-guide/#user-replaced-values